Modules/Segmentation/Watersheds/include/itkWatershedSegmenter.hxx

```diff
@@ -423,11 +424,8 @@ Segmenter<TInputImage>::InitializeBoundary()
       }
       this->GetBoundary()->GetFlatHash(idx)->clear();
       face = this->GetBoundary()->GetFace(idx);
-      faceIt = ImageRegionIterator<typename BoundaryType::face_t>(face, face->GetBufferedRegion());
-      for (faceIt.GoToBegin(); !faceIt.IsAtEnd(); ++faceIt)
-      {
-        faceIt.Set(fps);
-      }
+      const auto imageBufferRange = Experimental::MakeImageBufferRange(face.GetPointer());
+      std::fill(imageBufferRange.begin(), imageBufferRange.end(), fps);
```

Comment:
Or maybe just `face->FillBuffer(fps)`...?

Reply:
I expect that `FillBuffer` should be as fast or faster than `std::fill`.

Reply:
Assuming `face` is an image.

Reply:
> Assuming `face` is an image.

Yes it is  😄 `face`is a (smart) pointer to `face_t`, which is defined as: 

    using face_t = Image<face_pixel_t, TDimension>;

https://github.com/InsightSoftwareConsortium/ITK/blob/v5.1b01/Modules/Segmentation/Watersheds/include/itkWatershedBoundary.h#L117

`face->FillBuffer(fps)` is clearer and faster than the original code, but it's also simpler than `std::fill(imageBufferRange.begin(), imageBufferRange.end(), fps)`, I must admit. 😸